### PR TITLE
Corrects detect_python behavior (IDFGH-8832)

### LINKED ISSUE #10259: `detect_python.sh` fails to detect correct Python binary (IDFGH-8831)
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.1-dev-1908-g439a709c42

### Operating System used.

Linux

### How did you build your project?

Other (please specify in More Information)

### If you are using Windows, please specify command line type.

_No response_

### What is the expected behavior?

`detect_python.sh` should detect Python version 3.7

### What is the actual behavior?

`detect_python.sh` returns version 3.6.9

### Steps to reproduce.

On a Linux system with Python version with both 3.6.9 and 3.7 installed, attempt to install the idf. 

### Build or installation Logs.

```plain
Detecting the Python interpreter
Checking "python3" ...
Checking "python" ...
Python 3.6.9
"python" has been detected
Checking Python compatibility
Traceback (most recent call last):
  File "/home/appveyor/projects/simpleson-l4m5a/esp-idf/tools/python_version_checker.py", line 45, in <module>
    check()
  File "/home/appveyor/projects/simpleson-l4m5a/esp-idf/tools/python_version_checker.py", line 39, in check
    _ver_to_str(OLDEST_PYTHON_SUPPORTED), _ver_to_str(sys.version_info[:3])
RuntimeError: ESP-IDF supports Python 3.7 or newer but you are using Python 3.6.9. Please upgrade your installation as described in the documentation.
Setting IDF_PATH to '/home/appveyor/projects/simpleson-l4m5a/esp-idf'
Detecting the Python interpreter
Checking "python3" ...
Checking "python" ...
Python 3.6.9
"python" has been detected
Checking Python compatibility
Traceback (most recent call last):
  File "/home/appveyor/projects/simpleson-l4m5a/esp-idf/tools/python_version_checker.py", line 45, in <module>
    check()
  File "/home/appveyor/projects/simpleson-l4m5a/esp-idf/tools/python_version_checker.py", line 39, in check
    _ver_to_str(OLDEST_PYTHON_SUPPORTED), _ver_to_str(sys.version_info[:3])
RuntimeError: ESP-IDF supports Python 3.7 or newer but you are using Python 3.6.9. Please upgrade your installation
```

### More Information.

Full log can be found [here on Appveyor](https://ci.appveyor.com/project/gregjesl/simpleson-l4m5a/builds/45501913#L303)

The [Appveyor Linux build environment](https://www.appveyor.com/docs/linux-images-software/) has multiple versions of Python installed. Looking at `detect_python.sh`, the script should iterate through the versions until it finds a suitable version; however, the script is exiting with an unsupported version, as shown here in the logs:
```
Python 3.6.9
"python" has been detected
```
I suspect the problem is here: 
https://github.com/espressif/esp-idf/blob/439a709c4298b0f613d09b47fc0b7b9728fc5733/tools/detect_python.sh#L20

### PR DESCRIPTION
**Pull request by gregjesl**
Closes #10259

### CODE REVIEW

File: tools/detect_python.sh [hunk 1]
```diff
@@ -10,21 +10,30 @@
...
-
+    $p_cmd -c "import sys; exit(1) if sys.version_info.major < int(\"$OLDEST_PYTHON_SUPPORTED_MAJOR\") else exit(0);" || {
+        echo "\"$p_cmd\" has an unsupported major version"
+        continue
+    }
+    $p_cmd -c "import sys; exit(1) if sys.version_info.minor < int(\"$OLDEST_PYTHON_SUPPORTED_MINOR\") else exit(0);" || {
+        echo "\"$p_cmd\" has an unsupported minor version"
+        continue
+    }
+    echo "\"$p_cmd\" is a supported version"
```
Comment (ljden) [inline tools/detect_python.sh:30]: Extra print statements means output is different to `.fish` script. Also will add extra noise to log while installing - not sure if that's a good thing

**Review by ljden — COMMENTED** [review]
Just my 2c :)

**Comment by Alvin1Zhang** [comment]
Thanks for your contribution.

**Comment by gregjesl** [comment]
I've made my updates to conform to the styling standards, standing by for any feedback.

**Comment by gregjesl** [comment]
Based on the discussion in #10259, `python_version_checker.py` becomes obsolete and can be removed. The install script will need to be updated accordingly.

**Comment by dobairoland** [comment]
As per https://github.com/espressif/esp-idf/issues/10259#issuecomment-1329244449 I'm closing this one.

**Pull request closed without merging**